BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Paulo/PauloViewController.swift

```diff
@@ -6,13 +6,340 @@
     - How to customize it
  */
 
-
 import UIKit
+import MobileCoreServices
 
 class PauloViewController: UIViewController {
 
+    @IBOutlet weak private var tableView: UITableView!
+    private let refreshControl = UIRefreshControl()
+
+    private let viewModel = PauloViewModel()
+
     override func viewDidLoad() {
         super.viewDidLoad()
+        setupUI()
+        loadData()
+    }
+}
+
+// MARK: - Private Methods
+
+extension PauloViewController {
+
+    private func setupUI() {
+        title = viewModel.title
+
+        // Add the refresh control
+        tableView.addSubview(refreshControl)
+        refreshControl.tintColor = .red
+        refreshControl.addTarget(self, action: #selector(refreshData), for: .valueChanged)
+
+        // Register cells
+        let headerNib = UINib(nibName: String(describing: PauloHeaderView.self), bundle: nil)
+        tableView.register(headerNib, forHeaderFooterViewReuseIdentifier: PauloHeaderView.identifier)
+
+        let cellNib = UINib(nibName: String(describing: PauloTableViewCell.self), bundle: nil)
+        tableView.register(cellNib, forCellReuseIdentifier: PauloTableViewCell.identifier)
+
+        // Setting this gives us a slight performance boost, from the docs:
+        //   "using estimation allows you to defer some of the cost of geometry
+        //    calculation from load time to scrolling time."
+        tableView.estimatedRowHeight = PauloTableViewCell.estimatedRowHeight
```

Comment:
I recently posted on Slack an excellent talk about self sizing cells, did you watch it?

Reply:
I must have missed that!